components/br-stream/src/endpoint.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::fmt;
+use tokio::io::Result as TokioResult;
+use tokio::runtime::Runtime;
+use tokio_stream::StreamExt;
+
+use crate::metadata::store::MetaStore;
+use crate::metadata::{MetadataClient, MetadataEvent, Task as MetaTask};
+use crate::{errors::Result, observer::BackupStreamObserver};
+use online_config::ConfigChange;
+use raftstore::coprocessor::CmdBatch;
+use tikv::config::BackupConfig;
+use tikv_util::worker::{Runnable, Scheduler};
+use tikv_util::{debug, error, info};
+
+pub struct Endpoint<S: MetaStore + 'static> {
+    config: BackupConfig,
+    meta_client: MetadataClient<S>,
+    scheduler: Scheduler<Task>,
+    observer: BackupStreamObserver,
+    pool: Runtime,
+}
+
+impl<S> Endpoint<S>
+where
+    S: MetaStore + 'static,
+{
+    pub fn new(
+        meta_client: MetadataClient<S>,
+        config: BackupConfig,
+        scheduler: Scheduler<Task>,
+        observer: BackupStreamObserver,
+    ) -> Endpoint<S> {
+        let pool = create_tokio_runtime(config.num_threads, "br-stream")
+            .expect("failed to create tokio runtime for backup worker.");
+
+        // spawn a worker to watch task changes from etcd periodically.
+        let meta_client_clone = meta_client.clone();
+        let scheduler_clone = scheduler.clone();
+        pool.spawn(async move { Endpoint::starts_watch_tasks(meta_client_clone, scheduler_clone) });
```

Comment:
Maybe handle the error?

Reply:
we need a error handle mechanism to retry/reschedule the task in backup streaming.  I think we should finish that in future PR to make others not block on this PR.

I leave a TODO like `// TODO build a error handle mechanism #error 1`. that means we should consider to handle this error in mechanism, and this is the first error in mechanism.